### PR TITLE
Allow `Path` objects for deserialization of ONNX models

### PR DESCRIPTION
**Pull request by severinsimmler**
### Description
It is quite common to handle file paths with Python's `pathlib` module. The changes allow for deserialization of ONNX models from a `Path` object in addition to a file path string or a bytes object.

### Motivation and Context
In 8 out of 10 cases I forget that `InferenceSession` cannot handle `Path` objects and get a `TypeError: Unable to load from type '<class 'pathlib.PosixPath'>'`. There is actually no reason I can think of _not_ to allow `Path` objects. It should not have any side-effects since we cast that object to a string anyway.

### CODE REVIEW

**Comment by severinsimmler** [comment]
@microsoft-github-policy-service agree [company="{snapADDY GmbH}"]

**Comment by severinsimmler** [comment]
@microsoft-github-policy-service agree company=snapADDY GmbH

File: onnxruntime/python/onnxruntime_inference_collection.py
Comment (justinchuby) [inline onnxruntime/python/onnxruntime_inference_collection.py:9]: Consider importing modules only, and use pathlib.Path?

**Review by justinchuby — APPROVED** [review]
Will need code owner approval but thanks for making the change!

File: onnxruntime/test/python/onnxruntime_test_python.py
Comment (justinchuby) [inline onnxruntime/test/python/onnxruntime_test_python.py:12]: Same import modules only goes here

Comment (justinchuby) [reply onnxruntime/test/python/onnxruntime_test_python.py:12]: For the why:

Import only the module instead to keep the name space clean and provide readers with more context on where its members come from. For example, readers don’t need to backtrack to see `sleep` is a function from `time`, as opposed to a function defined in the file. https://google.github.io/styleguide/pyguide.html#22-imports

**Comment by severinsimmler** [comment]
Thanks, @justinchuby! Updated the import and fixed the unit test 👍

**Review by justinchuby — APPROVED** [review]

**Comment by justinchuby** [comment]
@severinsimmler could you merge from main? Thanks!

**Comment by severinsimmler** [comment]
@justinchuby :heavy_check_mark:

**Comment by hariharans29** [comment]
@justinchuby  - Is this PR good to go ?

**Pull request merged**
